lib/api/src/ptr.rs

```diff
@@ -103,7 +104,7 @@ impl<T: Copy + ValueType> WasmPtr<T, Item> {
     /// If you're unsure what that means, it likely does not apply to you.
     /// This invariant will be enforced in the future.
     #[inline]
-    pub fn deref<'a>(self, memory: &'a Memory) -> Option<&'a Cell<T>> {
+    pub fn deref<'a>(self, memory: &'a Memory) -> Option<WasmCell<T>> {
```

Comment:
Since we bind a lifetime explicitly here, we should probably set it in `WasmCell` here.. I'm not sure about the inference rules if `self` doesn't have a `&`, but when it does, it'd by default bind it to that. I'm guessing it's inferring unrelated lifetimes

Reply:
Fixed!